spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/wavefront/WavefrontProperties.java

```diff
@@ -0,0 +1,294 @@
+	}
+
+	/**
+	 * Returns the effective URI of the wavefront instance. This will not be the same URI
+	 * given through {@link #setUri(URI)} when a proxy is used.
+	 * @return the effective URI of the wavefront instance
+	 */
+	public URI getEffectiveUri() {
+		if (usesProxy()) {
+			// See io.micrometer.wavefront.WavefrontMeterRegistry.getWavefrontReportingUri
+			return URI.create(this.uri.toString().replace("proxy://", "http://"));
+		}
+		return this.uri;
+	}
+
+	/**
+	 * Returns the API token or throws an exception if the API token is mandatory. If a
+	 * proxy is used, the API token is optional.
+	 * @return the API token
+	 */
+	public String getApiTokenOrThrow() {
+		if (this.apiToken == null && !usesProxy()) {
+			throw new InvalidConfigurationPropertyValueException("management.wavefront.api-token", null,
+					"This property is mandatory whenever publishing directly to the Wavefront API");
+		}
+		return this.apiToken;
+	}
+
+	public String getSourceOrDefault() {
+		if (this.source != null) {
+			return this.source;
+		}
+		return getSourceDefault();
+	}
+
+	private String getSourceDefault() {
+		try {
+			return InetAddress.getLocalHost().getHostName();
+		}
+		catch (UnknownHostException ex) {
+			return "unknown";
+		}
+	}
+
+	private boolean usesProxy() {
+		return "proxy".equals(this.uri.getScheme());
+	}
+
+	public static class Sender {
+
+		/**
+		 * Maximum size of queued messages.
+		 */
+		private int maxQueueSize = 50000;
+
+		/**
+		 * Flush interval to send queued messages.
+		 */
+		private Duration flushInterval = Duration.ofSeconds(1);
+
+		/**
+		 * Maximum size of a message.
+		 */
+		private DataSize messageSize = DataSize.ofBytes(Integer.MAX_VALUE);
+
+		/**
+		 * Number of measurements per request to use for this backend. If more
+		 * measurements are found, then multiple requests will be made.
+		 */
+		private int batchSize = 10000;
+
+		public int getMaxQueueSize() {
+			return this.maxQueueSize;
+		}
+
+		public void setMaxQueueSize(int maxQueueSize) {
+			this.maxQueueSize = maxQueueSize;
+		}
+
+		public Duration getFlushInterval() {
+			return this.flushInterval;
+		}
+
+		public void setFlushInterval(Duration flushInterval) {
+			this.flushInterval = flushInterval;
+		}
+
+		public DataSize getMessageSize() {
+			return this.messageSize;
+		}
+
+		public void setMessageSize(DataSize messageSize) {
+			this.messageSize = messageSize;
+		}
+
+		public int getBatchSize() {
+			return this.batchSize;
+		}
+
+		public void setBatchSize(int batchSize) {
+			this.batchSize = batchSize;
+		}
+
+	}
+
+	public static class Metrics {
+
+		/**
+		 * Export configuration.
+		 */
+		private Export export = new Export();
+
+		public Export getExport() {
+			return this.export;
+		}
+
+		public void setExport(Export export) {
+			this.export = export;
+		}
+
+		public static class Export extends PushRegistryProperties {
+
+			/**
+			 * Global prefix to separate metrics originating from this app's white box
```

Comment:
Is white box a recognised Wavefront term? We try to avoid white box and black box if we can as they're jargon that can confuse people, particularly those with English as a second language.

Reply:
Ah, thanks for spotting this. I copied it from somewhere, gonna change this.

Reply:
I created https://github.com/spring-projects/spring-boot/issues/30649 to fix that in 2.5.x.